packages/core/src/common/keybinding.ts

```diff
@@ -144,27 +151,62 @@ export class KeybindingRegistry {
         }
     }
 
-    registerKeybindings(...bindings: Keybinding[]): void {
+    registerKeybindings(...bindings: (RawKeybinding | Keybinding)[]): void {
         for (const binding of bindings) {
             this.registerKeybinding(binding);
         }
     }
 
+    protected keybindingFromRaw(binding: RawKeybinding): Keybinding | undefined {
+        if (this.commandRegistry.getCommand(binding.command)) {
```

Comment:
Why is this check needed? Commands can be registered later, e.g. by language servers.

Reply:
Good point I copied it as it was in another function. I'll improve it now.

Reply:
Actually thinking more about this this means that we need to resolve the command later... before it was done in setKeyMap... but now we may need to store RawKeybindings in the registry and evaluate them only once they are looked-up.

I think I'll work a bit more on refactoring in this PR after all.